File: L2/ps2/readability.c
#include <cs50.h>
#include <stdio.h>
#include <string.h>

float count(string s);

int main(void)
{
    // Prompt the user for some text
    string text = get_string("Text: ");

    float index = count(text);

    // Print the Grade X
}

/**Count the L, the average number of letters per 100 words in the text
 * And the S, the average number of sentences per 100 words in the text
 * Then compute the Coleman-Liau index
 */
float count(string s)
{
    float L = 0, S = 0, letters = 0, words = 0, sentences = 0;
    for(int i = 0, n = strlen(s); i < strlen(s); i++)
    {
        if ((s[i] < 'z' && s[i] > 'a') || (s[i] < 'Z' && s[i] > 'A'))
        {
            letters++;
        }
        if(s[i] == ' ')
        {
            words++;
        }
        if(s[i] == '.' || s[i] == '?' || s[i] == '!' || s[i] == ',' || s[i] == ';' || s[i] == '.')
        {
            sentences++;
        }
    }
    printf("%f %f %f\n", letters, words, sentences);
    L = letters / words * 100;
    S = sentences / words *100;
    float index = 0.0588 * L - 0.296 * S -15.8;
    r
}

File: ps6/readability.py
text = input("Text: ")

# Compute the Coleman-Liau index
texts = text.lower()
# In Python, we can initialize variables like this, not in C
words, letters, sentences = 1, 0, 0
for c in texts:
    # Count the words
    if c == ' ':
        words += 1
    # Count the letters
    if c >='a' and c <= 'z':
        letters += 1
    # Count the sentences
    if c == '!' or c == '?' or c == '.':
        sentences += 1
# Final computation
L = letters / words * 100
S = sentences / words * 100
index = 0.0588 * L - 0.296 * S - 15.8

# Print the Grade
grade = round(index)
if grade < 1:
    print("Before Grade 1")
elif grade > 16:
    print("Grade 16+")
else:
    for i in range(1,16):
        if i == grade:
            print(f"Grade: {grade}")
